spring-cloud-starter-single-step-batch-job/src/main/java/org/springframework/cloud/task/batch/autoconfigure/flatfile/FlatFileItemReaderAutoConfiguration.java

```diff
@@ -84,24 +75,24 @@ public FlatFileItemReaderAutoConfiguration(FlatFileItemReaderProperties properti
 				.comments(this.properties.getComments()
 						.toArray(new String[this.properties.getComments().size()]));
 
-		if (this.lineTokenizer != null) {
-			mapFlatFileItemReaderBuilder.lineTokenizer(this.lineTokenizer);
+		if (lineTokenizer != null) {
+			mapFlatFileItemReaderBuilder.lineTokenizer(lineTokenizer);
```

Comment:
Why add the if statements? We previously depended upon the builder to validate that the state was valid.

Reply:
They were not required so , they can be null.   So we need the null check.

Reply:
Again...the builder does that check. We don't do null checks on any of the other places we do this kind of thing (see [here](https://github.com/spring-cloud/spring-cloud-task/pull/798/files#diff-af8f3bbf8f60a5583ee99cc0d35b7938d5838ccdcb4f8361a979fb4e16e96748R75) for example).

Reply:
Bye bye `if`s